### ISSUE TITLE
[WIP] Get scholars-test working on Beavernetes

### ISSUE THREAD

**Issue by decimalator** [osulp/Scholars-Archive#1919]
### Descriptive summary

As the first step towards moving staging to Beavernetes, we need to get sa-test up and running.

https://rancher.library.oregonstate.edu

**Comment by decimalator** [comment]
Most services have been spun up inside the `sa-test` namespace on the Beavernetes cluster. The `web` container needs a lot of the static assets from `shared` merged into it, so I'm working on an init-container to pre-populate it with the static assets 

🛠🛠🛠

**Comment by decimalator** [comment]
Blocked by #1980

**Comment by decimalator** [comment]
The new Beavernetes is setup and I've started re-creating sa-test.

`⇨ kubectl -n sa-test get pods                              02.25.20  17:01:18
NAME                         READY   STATUS    RESTARTS   AGE
blazegraph-758bbf56c-md89h   1/1     Running   0          34m
fcrepo-85d69bb88f-4jv84      1/1     Running   0          10m
memcache-854b558c6f-4qkhm    1/1     Running   0          89m
redis-588c9db777-wd42d       1/1     Running   0          8m10s`

**Comment by decimalator** [comment]
```
solr-76c974d5b4-j5pfk        1/1     Running       0          142m
```
web is ready as well, but I need to adjust the build process because it's geared towards the current prod/stage deployments, so none of the `shared` files are included (config/config.yml, etc.) So I'll need to get those merged into the build process so they make it into the containers too

**Comment by decimalator** [comment]
Working on converting the old YAML to the `apps/v1` format. 
Working on converting the base `Dockerfile` to use the `ruby:2.5-alpine` base container

**Comment by decimalator** [comment]
everything but the Rails and Sidekiq deployments are running in Beavernetes in the sa-test namespace. puma is giving me some grief, but the Rails piece is almost ready. 

- I've duplicated the scholars_stage database to scholars_test so we don't break the existing staging database. Like other systems, the database is still running on one of our MySQL VMs (lib-mdb1 since we can use MySQL 8)
- I haven't copied over the fcrepo data yet, but I wouldn't expect that to cause Rails to freak :thinking_face:
- The SOLR collection isn't setup yet, but the SolrCloud + ZooKeeper clusters are running
- The workers deployment isn't setup yet, that's my next step after fixing rails
- blazegraph is running, but I haven't tried copying its data from staging yet
- redis is running, but I haven't tried loading its data from staging yet
- memcached is running but that's quite possibly the easiest service in the world to run :joy:
- the Ingress isn't setup yet, we can use a test name like sa-test.l.o.e until we're ready to change ir-staging.l.o.e. 

while I'm debugging I've added a sleep loop at the end of the entrypoint.sh so I can tinker between iterations. I had to disable the liveness and readiness probes to let the container keep running, otherwise k8s restarts it, since puma never opens its listening port.

**Comment by decimalator** [comment]
- rails is up and running in "staging" mode. "staging" is pretty close to "production" atm
- workers is up and running in "staging" mode. it's processing the rake tasks, so seems to be working
- tools is up and running in "staging" mode. it's identical to rails and workers but just sleeps forever

I have a `solr:reindex` running to see if it will fill up the SOLR data from Fedora and MySQL, etc. It's still only to the point where it's talking to Fedora so I'll check on it tomorrow.

**Comment by decimalator** [comment]
The staging instance in Beavernetes is ready to test. Below is a list of URLs that exercise each of the various Rails controllers. 

- Verify that you can view each of these URLs without errors. 
- See if you can make changes to the item without errors.
- Try ingesting new works of various types.

https://sa-test.library.oregonstate.edu/collections/2z10wq369
https://sa-test.library.oregonstate.edu/collections/3n203z262
https://sa-test.library.oregonstate.edu/collections/x346d4386
https://sa-test.library.oregonstate.edu/concern/conference_proceedings_or_journals/0z708w53f
https://sa-test.library.oregonstate.edu/concern/graduate_projects/w9505068j
https://sa-test.library.oregonstate.edu/concern/articles/j9602083w
https://sa-test.library.oregonstate.edu/concern/articles/z890rt430
https://sa-test.library.oregonstate.edu/concern/defaults/wm117p117
https://sa-test.library.oregonstate.edu/concern/undergraduate_thesis_or_projects/qv33rw91s
https://sa-test.library.oregonstate.edu/concern/graduate_thesis_or_dissertations/bk128b10k
https://sa-test.library.oregonstate.edu/concern/honors_college_theses/zc77sq35t
https://sa-test.library.oregonstate.edu/concern/administrative_report_or_publications/rj430463s
https://sa-test.library.oregonstate.edu/concern/eesc_publications/sf268527j
https://sa-test.library.oregonstate.edu/concern/datasets/tm70mv47z
https://sa-test.library.oregonstate.edu/concern/technical_reports/w3763704q
https://sa-test.library.oregonstate.edu/concern/file_sets/nz805z82j
https://sa-test.library.oregonstate.edu/admin/admin_sets/jd472w57m
https://sa-test.library.oregonstate.edu/admin/stats
https://sa-test.library.oregonstate.edu/embargoes
https://sa-test.library.oregonstate.edu/admin/workflows
https://sa-test.library.oregonstate.edu/dashboard/collections
https://sa-test.library.oregonstate.edu/dashboard/works

**Comment by simholt** [comment]
All good for viewing!

**- Verify that you can view each of these URLs without errors.**

- [x] https://sa-test.library.oregonstate.edu/collections/2z10wq369 
- [x] https://sa-test.library.oregonstate.edu/collections/3n203z262 
- [x] https://sa-test.library.oregonstate.edu/collections/x346d4386 
- [x] https://sa-test.library.oregonstate.edu/concern/conference_proceedings_or_journals/0z708w53f 
- [x] https://sa-test.library.oregonstate.edu/concern/graduate_projects/w9505068j 
- [x] https://sa-test.library.oregonstate.edu/concern/articles/j9602083w 
- [x] https://sa-test.library.oregonstate.edu/concern/articles/z890rt430 
- [x] https://sa-test.library.oregonstate.edu/concern/defaults/wm117p117 
- [x] https://sa-test.library.oregonstate.edu/concern/undergraduate_thesis_or_projects/qv33rw91s 
- [x] https://sa-test.library.oregonstate.edu/concern/graduate_thesis_or_dissertations/bk128b10k 
- [x] https://sa-test.library.oregonstate.edu/concern/honors_college_theses/zc77sq35t 
- [x] https://sa-test.library.oregonstate.edu/concern/administrative_report_or_publications/rj430463s 
- [x] https://sa-test.library.oregonstate.edu/concern/eesc_publications/sf268527j 
- [x] https://sa-test.library.oregonstate.edu/concern/datasets/tm70mv47z 
	- Says in review but not listed in review queue, but looks same as on staging.
	- [x] Other dataset shows up fine: https://sa-test.library.oregonstate.edu/concern/datasets/m326m1935
- [x] https://sa-test.library.oregonstate.edu/concern/technical_reports/w3763704q 
	- locations not showing up - doesn't show up on regular staging, either, so is okay.
	- [x] Other technical report shows up fine: https://sa-test.library.oregonstate.edu/concern/technical_reports/gm80hv476
- [x] https://sa-test.library.oregonstate.edu/concern/file_sets/nz805z82j 
- [x] https://sa-test.library.oregonstate.edu/admin/admin_sets/jd472w57m 
- [x] https://sa-test.library.oregonstate.edu/admin/stats 
- [x] https://sa-test.library.oregonstate.edu/embargoes 
- [x] https://sa-test.library.oregonstate.edu/admin/workflows 
- [x] https://sa-test.library.oregonstate.edu/dashboard/collections 
- [x] https://sa-test.library.oregonstate.edu/dashboard/works

**Comment by simholt** [comment]
Only a couple issues, and I think they may be due to quirky test submissions that aren't playing by the rules.

**- See if you can make changes to the item without errors.**

- [x] https://sa-test.library.oregonstate.edu/collections/2z10wq369
- [x] https://sa-test.library.oregonstate.edu/collections/3n203z262
- [x] https://sa-test.library.oregonstate.edu/collections/x346d4386
- [x] https://sa-test.library.oregonstate.edu/concern/conference_proceedings_or_journals/0z708w53f
- [x] https://sa-test.library.oregonstate.edu/concern/graduate_projects/w9505068j
- [x] https://sa-test.library.oregonstate.edu/concern/articles/j9602083w
- [x] https://sa-test.library.oregonstate.edu/concern/articles/z890rt430
- [x] https://sa-test.library.oregonstate.edu/concern/defaults/wm117p117
- [x] https://sa-test.library.oregonstate.edu/concern/undergraduate_thesis_or_projects/qv33rw91s
- [x] https://sa-test.library.oregonstate.edu/concern/graduate_thesis_or_dissertations/bk128b10k
- [x] https://sa-test.library.oregonstate.edu/concern/honors_college_theses/zc77sq35t
- [x] https://sa-test.library.oregonstate.edu/concern/administrative_report_or_publications/rj430463s
- [x] https://sa-test.library.oregonstate.edu/concern/eesc_publications/sf268527j
- [ ] https://sa-test.library.oregonstate.edu/concern/datasets/tm70mv47z (deleted this problem file -- another dataset test worked fine: 	- [x] https://sa-test.library.oregonstate.edu/concern/datasets/m326m1935 )
- [x] https://sa-test.library.oregonstate.edu/concern/technical_reports/w3763704q
- [x] https://sa-test.library.oregonstate.edu/concern/file_sets/nz805z82j
- [x] https://sa-test.library.oregonstate.edu/admin/admin_sets/jd472w57m
- [ ] https://sa-test.library.oregonstate.edu/admin/stats
	- [ ] updates as new items are added but get broken message when set date limiters
- [x] https://sa-test.library.oregonstate.edu/embargoes
- [x] https://sa-test.library.oregonstate.edu/admin/workflows
- [x] https://sa-test.library.oregonstate.edu/dashboard/collections
- [ ] https://sa-test.library.oregonstate.edu/dashboard/works
	- [ ] can't get to last page under All Works - might be due to janky test files

**Comment by simholt** [comment]
**- Try ingesting new works of various types.**

- [x] Technical report
- [ ] Default/Other - uploaded but skipped review queue, will write up a ticket
- [x] EESC
- [x] Admin report
- [x] ETD
- [x] HCT
- [x] Article
- [ ] bulk upload - worked although duplicated them as empty files, which I thought we'd fixed?

**Comment by CGillen** [comment]
```
 Default/Other - uploaded but skipped review queue, will write up a ticket
```
This is a misconfiguration between staging and production


Staging:
<img width="1157" alt="image" src="https://user-images.githubusercontent.com/11052958/134231438-9a85d110-197a-480d-a744-1fb8c6f251be.png">
Prod:
<img width="1157" alt="image" src="https://user-images.githubusercontent.com/11052958/134231459-9f087989-92ad-494f-8660-4a15abb81e80.png">

**Comment by CGillen** [comment]
`tm70mv47z` is missing from Fedora. I doubt this is specific to the cluster, more likely an existing issue

**Comment by CGillen** [comment]
`https://sa-test.library.oregonstate.edu/admin/stats` breaks on prod and staging when entered directly without `?local=en`.
For some reason the form submit doesn't add the local back and it causes breakage.
`https://ir.library.oregonstate.edu/admin/stats?utf8=%E2%9C%93&locale=en&stats_filters%5Bstart_date%5D=2021-09-15&stats_filters%5Bend_date%5D=&commit=Load+Stats`
vs
`https://ir.library.oregonstate.edu/admin/stats?utf8=%E2%9C%93&stats_filters%5Bstart_date%5D=2021-09-15&stats_filters%5Bend_date%5D=&commit=Load+Stats`

**Comment by CGillen** [comment]
`bulk upload - worked although duplicated them as empty files, which I thought we'd fixed?`
This is yet another misconfiguration between staging and production =.=
The admin user on production has the username `admin` and the `User.batch_user_key` and `User.audit_user_key` set to `:username`. This allowed batch operations to send a message to `scholarsarchive@oregonstate.edu`.
Staging's admin user had the username `scholarsarchive` with `User.batch_user_key = batchuser@example.com` and `User.audit_user_key = admin`, so when the messenger went to send the completion message it couldn't find a user to send it to. This caused a crash in the job that creates the work, which caused the job to requeue and create more works (this time without files)
